### PR TITLE
Implement difference operation on primitive sets. Partially closes #310.

### LINKED ISSUE #310: Implement classic set operations on primitive sets.
Implement the following set operations on primitive sets, to be symmetric with the same forms on SetIterable:

```
<type>Set union(<type>Set);
<type>Set intersect(<type>Set);
<type>Set difference(<type>Set);
<type>Set symmetricDifference(<type>Set);
boolean isSubSetOf(<type>Set);
boolean isProperSubsetOf(<type>Set);
LazyIterable<<type><type>Pair> cartesianProduct(<type>Set);
```

### PR DESCRIPTION
**Pull request by prathasirisha**
Signed-off-by: Sirisha Pratha <sirisha.pratha@bnymellon.com>

### CODE REVIEW

File: unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/AbstractBooleanSetTestCase.java [hunk 1]
```diff
@@ -893,4 +893,38 @@ public void intersect()
...
+        MutableBooleanSet set14 = this.setWithTrueFalse;
+        MutableBooleanSet set24 = this.newWith();
+        MutableBooleanSet actual4 = set14.difference(set24);
+        Assert.assertEquals(this.setWithTrueFalse, actual4);
+
+        MutableBooleanSet set15 = this.newWith();
+        MutableBooleanSet set25 = this.setWithTrueFalse;
+        MutableBooleanSet actual5 = set15.difference(set25);
+        Assert.assertEquals(this.emptySet, actual5);
+
```
Comment (donraab) [inline unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/AbstractBooleanSetTestCase.java:924]: I think we just need to add a case where you have `this.setWithTrueFalse` and call `difference` with `this.setWithTrueFalse`.

File: unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/ImmutableBooleanHashSetTest.java [hunk 2]
```diff
@@ -625,4 +625,38 @@ public void intersect()
...
+        ImmutableBooleanSet actual3 = set13.difference(set23);
+        Assert.assertEquals(this.emptySet, actual3);
+
+        ImmutableBooleanSet set14 = this.trueFalseSet;
+        ImmutableBooleanSet set24 = this.newWith();
+        ImmutableBooleanSet actual4 = set14.difference(set24);
+        Assert.assertEquals(this.trueFalseSet, actual4);
+
+        ImmutableBooleanSet set15 = this.newWith();
+        ImmutableBooleanSet set25 = this.trueSet;
```
Comment (donraab) [inline unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/ImmutableBooleanHashSetTest.java:653]: In the mutable test, I believe you used `this.trueFalseSet` for this test. We should keep the symmetry in this test.

File: unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/ImmutableBooleanHashSetTest.java [hunk 3]
```diff
@@ -625,4 +625,38 @@ public void intersect()
...
+        ImmutableBooleanSet actual5 = set15.difference(set25);
+        Assert.assertEquals(this.emptySet, actual5);
+
```
Comment (donraab) [inline unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/ImmutableBooleanHashSetTest.java:656]: I think we just need to add a case where you have `this.trueFalseSet` and call `difference` with `this.trueFalseSet.`

**Review by donraab — CHANGES_REQUESTED** [review]
Hi @prathasirisha, I've commented on a few small changes. Other than those, I think this looks good. Thank you!

File: unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/ImmutableBooleanHashSetTest.java (hunk 3 above)
Comment (prathasirisha) [reply unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/ImmutableBooleanHashSetTest.java:656]: Updated, see lines 642-643.

File: unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/AbstractBooleanSetTestCase.java (hunk 1 above)
Comment (prathasirisha) [reply unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/AbstractBooleanSetTestCase.java:924]: Updated, see lines 910-911.

File: unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/ImmutableBooleanHashSetTest.java (hunk 2 above)
Comment (prathasirisha) [reply unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/ImmutableBooleanHashSetTest.java]: Done.

**Review by donraab — APPROVED** [review]
LGTM.

**Pull request merged**